Java/Java.sublime-syntax

```diff
@@ -67,47 +85,134 @@ contexts:
           scope: punctuation.separator.property.java
     - match: '@\w*'
       scope: storage.type.annotation.java
+
   anonymous-classes-and-new:
-    - match: \b(?<!::)new\b
+    - match: \bnew\b
       scope: keyword.control.new.java
+      push: instantiation
+
+  instantiation:
+    - match: '{{primitives}}'
+      scope: storage.type.primitive.java
+      set: instantiation-arrays
+    - match: '{{qualified_id}}'
```

Comment:
Missed this in my initial scan of your review…

This would be a trickier change, since we're nesting `{{qualified_id}}` matches within *other* lookahead contexts (for field matching).  It's not impossible, and I agree it's the right thing to do, it's just probably going to be very messy.  If it's alright with you, I'd rather tilt that windmill in a separate PR, since it's going to require some deeper changes.